src/extension/geofence/src/main/java/org/geoserver/geofence/ContainerLimitResolver.java

```diff
@@ -0,0 +1,492 @@
+        }
+        if (noneProcess != null) {
+            intersectArea =
+                    unionOrReturnIfNull(
+                            () -> noneProcess.getIntersectArea(), intersectArea, lessRestrictive);
+            clipArea =
+                    unionOrReturnIfNull(() -> noneProcess.getClipArea(), clipArea, lessRestrictive);
+            catalogModeDTO = getLarger(catalogModeDTO, noneProcess.getCatalogModeDTO());
+        }
+        return new ProcessingResult(intersectArea, clipArea, catalogModeDTO);
+    }
+
+    private Geometry unionOrReturnIfNull(
+            Supplier<Geometry> supplier, Geometry area, boolean lessRestrictive) {
+        if (area != null)
+            area = this.areasHelper.reprojectAndUnion(supplier.get(), area, lessRestrictive);
+        else area = supplier.get();
+        return area;
+    }
+
+    // enlarge a processing result
+    private ProcessingResult enlargeGroupProcessingResult(List<ProcessingResult> processingResult) {
+        if (processingResult == null || processingResult.isEmpty()) return null;
+        CatalogModeDTO catalogModeDTO = null;
+        Geometry intersectArea = null;
+        Geometry clipArea = null;
+        boolean lessRestrictive = groupSummaries != null;
+        for (int i = 0; i < processingResult.size(); i++) {
+            ProcessingResult pr = processingResult.get(i);
+            catalogModeDTO = getLarger(catalogModeDTO, pr.getCatalogModeDTO());
+            Geometry allowedArea = pr.getIntersectArea();
+            Geometry allowedAreaClip = pr.getClipArea();
+            if (i == 0) {
+                intersectArea = allowedArea;
+                clipArea = allowedAreaClip;
+            } else {
+                intersectArea =
+                        areasHelper.reprojectAndUnion(intersectArea, allowedArea, lessRestrictive);
+                clipArea =
+                        areasHelper.reprojectAndUnion(clipArea, allowedAreaClip, lessRestrictive);
+            }
+        }
+
+        return new ProcessingResult(intersectArea, clipArea, catalogModeDTO);
+    }
+
+    /**
+     * Resolve all the AccessInfo beloging to the same role restricting the access for allowed areas
+     * (intersection) and catalog mode.
+     *
+     * @param resourceAccessInfo the resource access infos by role
+     * @param groupAccessInfoByRole groups access infos by role.
+     * @return a Multimap<String,AccessInfo> gathering all the AccessInfo grouped by role.
+     */
+    private ListMultimap<RestrictionType, ProcessingResult> intersectAccesses(
+            Map<String, AccessInfo> resourceAccessInfo,
+            ListMultimap<String, AccessInfo> groupAccessInfoByRole) {
+        ListMultimap<RestrictionType, ProcessingResult> multiMap = ArrayListMultimap.create();
+        for (String key : resourceAccessInfo.keySet()) {
+            @SuppressWarnings("unchecked")
+            List<AccessInfo> infos = groupAccessInfoByRole.get(key);
```

Comment:
Is this suppression still needed? Does not look like it would.

Reply:
Removed it as well, thanks!